cfe_internal/recommendations.cf

```diff
@@ -0,0 +1,66 @@
+bundle agent postgresql_conf_reccomendations
+# @brief Recommendations about the configuration of postgresql.conf for CFEngine Enterprise Hubs
+{
+  meta:
+
+   (policy_server|am_policy_hub).enterprise_edition::
+
+    "tags" slist => { "cfengine_recommends" };
+
+  vars:
+    "pgsql_conf" string => "/var/cfengine/state/pg/data/postgresql.conf";
+    "pgsql_conf" string => "/tmp/postgresql.conf";
+    "mem_info_source" string => "/proc/meminfo";
+    "mem_info_data" data => data_readstringarray( $(mem_info_source), "", "(:|\s+)", inf, inf);
+
+    "upper" string => "67108864"; # 64 * 1024 * 1024 in KB
+    "lower" string => "3145728"; # 3 * 1024 * 1024 in KB
+
+      "conf[maintenance_work_mem]"
+        string => "2GB",
+        if => isgreaterthan( "$(mem_info_data[MemTotal][1])", $(lower) ),
+        comment => "If we have more than 2GB of memory available then we set the
+        maintenance_work_memory to 2G to improve index creation, and
+        vacuuming. Else we leave the default value.";
+
+      "conf[shared_buffers]"
+        string => "16GB",
+        if => isgreaterthan( "$(mem_info_data[MemTotal][1])", $(upper) );
+
+      "conf[effective_cache_size]"
+        string => "11GB", # 70% of 16GB
+        if => isgreaterthan( "$(mem_info_data[MemTotal][1])", $(upper) );
+
+      "calculated_shared_buffers_MB"
+        string => format( "%d0",
```

Comment:
What does the `0` at the end of the format string mean?

Reply:
it formats it to an integer with 0 decimals.